### PR TITLE
Add dropdown, radio, slider, and text field parts

### PR DESCRIPTION
**Pull request by brett-groshong**
### Purpose of this PR
This change adds the dropdown, radio, slider, and text fields into available node parts.

---
### Testing status
Manual testing steps:
1) Open a shader graph
2) Right-click and select "Create Node"
Notice: TestUISlider node is available.

Screen Shot 2022-03-04 at 9 47 41 AM

---
### Comments to reviewers

*Dropdown*
- DropdownPart was added but is not being used in an example because it should be refactored as listed in the DropdownPart code.

*TextField*
- TextField was added as a UI Part but should be removed.
- The TextField is specifically for the Swizzle node that will be added when custom nodes and dynamic topologies are being added.

### CODE REVIEW

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs [hunk 1]
```diff
@@ -134,4 +134,20 @@ internal class TestUIColorRGBANode : IStandardNode
...
+            1, // Version
+            "TestUISlider", // Name
+            "Out = In;",
+            new ParameterDescriptor("In", TYPE.Float, GraphType.Usage.Static),
+            new ParameterDescriptor("Out", TYPE.Float, GraphType.Usage.Out)
+        );
+
+        public static Dictionary<string, float> UIHints => new()
+        {
+            { "In.UseSlider", 1 } // Use a slider for In (float value is ignored)
```
Comment (esmelusina) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs:150]: Don't sliders need a range?

File: com.unity.shadergraph/Editor/GraphUI/GraphElements/ModelUI/StaticPortParts/RadioPart.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+using UnityEditor.ShaderGraph.Registry;
+using UnityEditor.ShaderGraph.Registry.Types;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ShaderGraph.GraphUI
+{
+    public class RadioPart : SingleFieldPart<RadioButtonGroup, int>
+    {
+        protected override string UXMLTemplateName => "StaticPortParts/SliderPart";
+        protected override string FieldName => "sg-slider";
```
Comment (esmelusina) [inline com.unity.shadergraph/Editor/GraphUI/GraphElements/ModelUI/StaticPortParts/RadioPart.cs:12]: RadioButtonGroup is used for SliderPart?

Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphUI/GraphElements/ModelUI/StaticPortParts/RadioPart.cs:12]: Oops. Thanks. 
Updating

Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphUI/GraphElements/ModelUI/StaticPortParts/RadioPart.cs:12]: I assume that there will be other errors of this format but I will change them all.

**Review by esmelusina — CHANGES_REQUESTED** [review]

File: com.unity.shadergraph/Editor/GraphUI/GraphElements/ModelUI/StaticPortParts/RadioPart.cs (hunk 2 above)
Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphUI/GraphElements/ModelUI/StaticPortParts/RadioPart.cs]: Updated.

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs (hunk 1 above)
Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs:150]: I will put up another PR for adding options to the UI hints. There are more places than just this where we'd want them.

**Review by dhsavell — APPROVED** [review]

**Review by esmelusina — APPROVED** [review]
Looks good!

**Pull request merged**